src/figuresClasses.ts

```diff
@@ -1,19 +1,103 @@
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
+
+enum Color {
+  Red = 'red',
+  Green = 'green',
+  Blue = 'blue'
+}
+
+const MAIN_ERROR_MESSAGE = 'Invalid values';
+const ERROR_MESSAGE_TRIANGLE = `${MAIN_ERROR_MESSAGE}:
+  the longest side of a triangle is >= than a sum of two others`;
+const ERROR_MESSAGE_CIRCLE = `${MAIN_ERROR_MESSAGE}:
+  radius of a circle is smaller than 0`;
+const ERROR_MESSAGE_RECTANGLE = `${MAIN_ERROR_MESSAGE}:
+  width or height of a rectangle is smaller than 0`;
+
+const handleRound = (square: number): number => Math.floor(square * 100) / 100;
+const isTriangleValid = (a: number, b: number, c: number): boolean => {
+  return a + b <= c || a + c <= b || c + b <= a || a <= 0 || b <= 0 || c <= 0;
+};
+
 export interface Figure {
+  shape: Shape;
+  color: Color;
 
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape = Shape.Triangle;
 
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (isTriangleValid(a, b, c)) {
+      throw new Error(ERROR_MESSAGE_TRIANGLE);
```

Comment:
If the problem is that the number is less than 0, then the error text will be irrelevant.